tests/test_abstract_repr.py

```diff
@@ -228,27 +229,75 @@ def _roundtrip(abstract_device):
 
     def test_exceptions(self, abstract_device):
         def check_error_raised(
-            obj_str: str, original_err: Type[Exception], err_msg: str = ""
+            obj_str: str,
+            original_err: Type[Exception],
+            err_msg: str = "",
+            check_from_abstract_repr=False,
         ) -> Exception:
-            with pytest.raises(DeserializeDeviceError) as exc_info:
-                deserialize_device(obj_str)
-
-            cause = exc_info.value.__cause__
-            assert isinstance(cause, original_err)
-            assert re.search(re.escape(err_msg), str(cause)) is not None
+            func_to_test = [deserialize_device]
+            err_raised = []
+            if check_from_abstract_repr:
+                func_to_test += [
+                    Device.from_abstract_repr,
+                    VirtualDevice.from_abstract_repr,
+                ]
+            for func in func_to_test:
+                with pytest.raises(DeserializeDeviceError) as exc_info:
+                    func(obj_str)
+                err_raised.append(exc_info)
+            for exc_info in err_raised:
+                cause = exc_info.value.__cause__
+                assert isinstance(cause, original_err)
+                assert re.search(re.escape(err_msg), str(cause)) is not None
```

Comment:
This is looking unnecessarily convoluted. Why not just make the function to test an argument to the function (potentially with the default being deserialize_device) and then do the test for the from_abstract_repr() outside of the function?